### ISSUE TITLE
Wide Glyphs do not work correctly on `conhost`

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#2943]
Run the scenario `Buttons` with host console and you'll see the same. I really don't know why this happens but I'm sure the right string are passed to the console.

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/244520d8-57c0-41ee-bd62-6956c57a6402)

_Originally posted by @BDisp in https://github.com/gui-cs/Terminal.Gui/issues/2928#issuecomment-1785413495_
            
This problem is not just about `Button`. `conhost` operates differently from WT when rendering wide glyphs. 

Run CharMap under conhost to see:
![image](https://github.com/gui-cs/Terminal.Gui/assets/585482/5692c46c-27c1-4de2-8ede-9861d47fc907)

Note that `NetDriver` doesn't work correctly either.:
![image](https://github.com/gui-cs/Terminal.Gui/assets/585482/29f467ec-eb4a-40dc-8b49-330214d2c844)

**Comment by tig** [comment]
I am beginning to suspect the REAL problem is `Rune.GetColumns()`, which depends on `wcwidth` is returning values different than what WT is using internally (and WT an conhost behave differently in this regard).

The real fix to all of this is to figure out how to calculate the number of columns the same way WT is doing it.

**Comment by tig** [comment]
> Yes, this probably be a good choice. Do you are going to investigate the code?

Yes, I am actively working on this as part of #2610

**Comment by BDisp** [comment]
I think that haven't anything related with wide char. I did this change to `GetColumns`:

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/f35ab94f-ae48-4f1a-9f9b-fc87c396d25b)

and the result is:

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/eadf721b-bfc2-4a45-911d-e6027e595122)

The width is really 1 but `conhost` doesn't render well on all situations with certain character.

**Comment by tig** [comment]
It's more subtle than that. How each terminal internally stores code points in cells and how it deals with cells to the left or right varies widely. 

Run v1 charmap in conhost/wt/netdriver/windowsdriver and see wild differences.

**Comment by BDisp** [comment]
> It's more subtle than that. How each terminal internally stores code points in cells and how it deals with cells to the left or right varies widely.
> 
You are right is terminal dependent. The case of the `⟦` the terminal doesn't handle the same way for all of them. In this case is only on the second button of the Top app and on the Dialog the situation is worst.

> Run v1 charmap in conhost/wt/netdriver/windowsdriver and see wild differences.

Yes I already did and I see it happens with `WindowsDriver` and `NetDriver`. We can't do nothing unless it's fixed by Microsoft I guess.

**Comment by BDisp** [comment]
@tig see this code point U+1085F “𐡟” where seems to occupies two columns but `GetColumn` return one. I know that depend if the font support or not a code point but `cmd` and `conhost` support it. It feels like it gets too tight.

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/bdf985f5-d035-43f6-b304-d41ccef65028)

**Comment by tig** [comment]
@BDisp this will be fixed in #4421 right?

**Comment by BDisp** [comment]
> [@BDisp](https://github.com/BDisp) this will be fixed in [#4421](https://github.com/gui-cs/Terminal.Gui/pull/4421) right?

Not for all situations. The button brackets only can be achieved with a supported font. I'm using "DejaVuSansM Nerd Font Mono" but the are also more supported fonts.

<img width="444" height="307" alt="Image" src="https://github.com/user-attachments/assets/76f6dff4-4a91-4dc3-bf2d-42b1c08b2d84" />

For wide chars and surrogate pairs it is working as expected. But for ZWJ (Zero Width Joiner) it is not possible to display the correct grapheme because it's a console limitation. For example, for supported terminal the grapheme 👨‍👩‍👧‍👦 is displayed correctly, but in cmd.exe or conhost.exe it can't render, even we output it as string to the console. So, it's displayed as below in the last line:

<img width="1207" height="595" alt="Image" src="https://github.com/user-attachments/assets/c9042205-eecd-4dd9-8f8e-f2f59e63ce6d" />

But isn't a Terminal.Gui fault but only a support limitation on stale consoles.

**Comment by tig** [comment]
Can you propose a better title for this Issue that identifies the remaining problem?

**Comment by BDisp** [comment]
> Can you propose a better title for this Issue that identifies the remaining problem?

I think that the remaining problem for ZWJ graphemes aren't soluble, a least with the current API, unless an improvement will be done, which I don't believe,
See the comparation between the images in the https://github.com/gui-cs/Terminal.Gui/issues/2943#issue-1968769662  comment above and the the below images, which are aligned without screwup the scroll bar.
In my opinion this issue can be closed or I can add this issue to my PR to close it later.

<img width="1205" height="648" alt="Image" src="https://github.com/user-attachments/assets/c55c387a-8499-4cfa-95d5-41980d57318f" />

<img width="1101" height="638" alt="Image" src="https://github.com/user-attachments/assets/c15a1f5c-7261-4523-ac68-3bbd64bfc5d8" />
